### PR TITLE
LibRegex: Fix capture group clearing issues / LibRegex: Fix parsing of /a{/ and friends

### PR DESCRIPTION
**Pull request by alimpfard**

### CODE REVIEW

File: Userland/Libraries/LibRegex/RegexParser.cpp [hunk 1]
```diff
@@ -1153,6 +1153,9 @@ bool ECMA262Parser::parse_quantifier(ByteCode& stack, size_t& match_length_minim
         auto low_bound = low_bound_string.to_uint();
 
         if (!low_bound.has_value()) {
+            if (done())
+                return set_error(Error::MismatchingBrace);
```
Comment (Lubrsi) [inline Userland/Libraries/LibRegex/RegexParser.cpp:1157]: This isn't an error, it just becomes a literal.
For example:
Screenshot from 2021-07-24 10-49-56

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexParser.cpp:1157]: Oh.
Then the previous should've been correct, hmmmm

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexParser.cpp:1157]: Looks like that's actually an annex B thing, the base ECMA262 grammar does not allow that.
See https://tc39.es/ecma262/#prod-annexB-ExtendedPatternCharacter vs. https://tc39.es/ecma262/#prod-PatternCharacter

So the error is correct, but only if we're not using the browser extended grammar.

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexParser.cpp:1157]: Updated to allow it in annex B mode.

**Pull request merged**